### ISSUE TITLE
Breaking the repeat command

### ISSUE THREAD

**Issue by davidmh** [metakirby5/codi.vim#37]
I'm not sure if this is just an incompatibility problem with [vim-repeat](https://github.com/tpope/vim-repeat) or the repeat command in general.

Sometimes codi can't replicate the last operation with the dot command.

I can reproduce the issue everytime using [vim-unimpaired](https://github.com/tpope/vim-unimpaired)'s `]e` to swap lines

With codi off:
https://gfycat.com/VapidUnfitCopperhead

With codi on:
https://gfycat.com/PoliticalSereneBettongkk

But it has happened with when replacing matches with `cgn`, instead of replacing the match, it deletes a couple of lines.

**Comment by metakirby5** [comment]
Thanks for the report. I've just pushed up a fix for `g:codi#sync` (I had inadvertently broken it. I'll write tests someday!), so would you mind updating your `Codi` install and trying `]e` with `vim -c 'let g:codi#sync = 1 | Codi' filename.py`? Synchronous should be compatible. Asynchronous behavior still seems broken, but I'm looking into it.

**Comment by davidmh** [comment]
Yes, it does solve the problem, it moves the cursor to some odd positions, but it's completely functional. Thank you.

Great plugin, btw!

**Comment by metakirby5** [comment]
What do you mean by moves the cursor to some odd positions? Are you sure you have the latest version? I believe I know what you're talking about but I thought I had just fixed that.

**Comment by davidmh** [comment]
You're right, looks like I only thought about updating the plugin but I obviously didn't.

I did find another problem, this time with the jump commands `CTRL-O` and `CTRL-I`. They create empty buffers between jumps.

https://gfycat.com/WaterloggedElasticKarakul

Could it be related?

**Comment by metakirby5** [comment]
Unfortunately, I haven't found a way of buffer switching without adding it to the jumplist. I tried `keepjumps`, but it didn't seem to work. The ideal solution would be to update the Codi buffer without switching buffers at all, but I don't know of a way to write to a buffer other than the one you are on. If you have any suggestions, please let me know :)
